A32NX/html_ui/Pages/A32NX_Utils/SimBriefApi.js

```diff
@@ -1,17 +1,35 @@
 class SimBriefApi {
-    static getSimBriefOfp(username, userId) {
-        let apiUrl;
-        if (username) {
-            apiUrl = `${SimBriefApi.url}&username=${username}`;
-        } else if (userId) {
-            apiUrl = `${SimBriefApi.url}&userid=${userId}`;
+    static getSimBriefOfp(userId) {
+        if (userId) {
+            return fetch(`${SimBriefApi.url}&userid=${userId}`)
+                .then((response) => {
+                    if (!response.ok) {
+                        throw (response);
+                    }
+
+                    return response.json();
+                });
         } else {
-            throw ("No SimBrief username/user ID provided");
+            throw ("No SimBrief pilot ID provided");
```

Comment:
This would usually just be written like this:

```suggestion
            throw new Error("No SimBrief pilot ID provided");
```